src/library/scala/collection/immutable/LazyList.scala

```diff
@@ -580,67 +527,65 @@ sealed private[immutable] trait LazyListFactory[+CC[+X] <: LinearSeq[X] with Laz
   * @define Coll `LazyList`
   */
 @SerialVersionUID(3L)
-object LazyList extends LazyListFactory[LazyList] {
-
-  protected def newCons[T](hd: => T, tl: => LazyList[T]): LazyList[T] = new LazyList.Cons(hd, tl)
+object LazyList extends SeqFactory[LazyList] {
+  private[this] val _empty = newLL(State.Empty)
```

Comment:
IMHO, the fact that the empty state is evaluated once and cached for other lazy lists makes things harder to reason about. I think we should `force` it at definition site.